### PR TITLE
Handle only those http requests that equal to adjusted websocket path

### PR DESCRIPTION
**Pull request by jamel**
Motivation:

It will be easier to support websockets in server application by using WebSocketServerProtocolHandshakeHandler class and not reinvent its functionality. But currently it handles all http requests as if they were websocket handshake requests.

Modifications:

Check if http request path is equals to adjusted websocket path.
Fixed example of websocket server implementation.

Result:

WebSocketServerProtocolHandshakeHandler handles only websocket handshake requests.

### CODE REVIEW

**Comment by normanmaurer** [comment]
@jamel please fix check style errors.

File: example/src/main/java/io/netty/example/http/websocketx/server/WebSocketFrameHandler.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+public class WebSocketFrameHandler extends SimpleChannelInboundHandler<WebSocketFrame> {
+
+    @Override
+    protected void channelRead0(ChannelHandlerContext ctx, WebSocketFrame frame) throws Exception {
+        // ping and pong frames already handled
+
+        if (frame instanceof TextWebSocketFrame) {
+            // Send the uppercase string back.
+            String request = ((TextWebSocketFrame) frame).text();
+            System.err.printf("%s received %s%n", ctx.channel(), request);
```
Comment (netkins) [inline example/src/main/java/io/netty/example/http/websocketx/server/WebSocketFrameHandler.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS106)

File: example/src/main/java/io/netty/example/http/websocketx/server/WebSocketFrameHandler.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            ctx.channel().writeAndFlush(new TextWebSocketFrame(request.toUpperCase()));
```
Comment (netkins) [inline example/src/main/java/io/netty/example/http/websocketx/server/WebSocketFrameHandler.java]: MAJOR When doing a String.toLowerCase()/toUpperCase() call, use a Locale [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AUseLocaleWithCaseConversions)

**Comment by jamel** [comment]
@normanmaurer check style errors are fixed. Also squashed all commits into one.

File: example/src/main/java/io/netty/example/http/websocketx/server/WebSocketFrameHandler.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        // ping and pong frames already handled
+
+        if (frame instanceof TextWebSocketFrame) {
+            // Send the uppercase string back.
+            String request = ((TextWebSocketFrame) frame).text();
+            logger.info("{} received {}", ctx.channel(), request);
+            ctx.channel().writeAndFlush(new TextWebSocketFrame(request.toUpperCase(Locale.US)));
+        } else {
+            String message = String.format("%s frame types not supported", frame.getClass().getName());
+            throw new UnsupportedOperationException(message);
```
Comment (trustin) [inline example/src/main/java/io/netty/example/http/websocketx/server/WebSocketFrameHandler.java]: How about:

    "unsupported frame type: " + frame.getClass().getName()

Comment (jamel) [inline example/src/main/java/io/netty/example/http/websocketx/server/WebSocketFrameHandler.java]: fixed

**Comment by normanmaurer** [comment]
@jamel please squash

**Comment by normanmaurer** [comment]
Cherry-picked into 4.0 (2e6544fc0e2581f96652be02d86e777e8e0493bc)  and 4.1(68bbd4e9669aab4c29dcc1d23ef331aaa2aee333)

**Pull request closed without merging**